fdb-extensions/src/main/java/com/apple/foundationdb/async/RTree.java

```diff
@@ -0,0 +1,1878 @@
+            if (hilbertValueCompare >= 0) {
+                // child HV >= target HV
+                if (hilbertValueCompare == 0) {
+                    // child == target
+                    final var tupleCompare = TupleHelpers.compare(slot.getKey(), key);
+                    if (tupleCompare >= 0) {
+                        return slotIndex;
+                    }
+                } else {
+                    return slotIndex;
+                }
+            }
+        }
+
+        // this is an intermediate node; we insert INTO the last child
+        return slots.size() - 1;
+    }
+
+    private static int findInsertItemSlotIndex(@Nonnull final LeafNode leafNode,
+                                               @Nonnull final BigInteger hilbertValue,
+                                               @Nonnull final Tuple key) {
+        final List<ItemSlot> items = leafNode.getItems();
+        for (int slotIndex = 0; slotIndex < items.size(); slotIndex++) {
+            final ItemSlot slot = items.get(slotIndex);
+            final int hilbertValueCompare = slot.getHilbertValue().compareTo(hilbertValue);
+            if (hilbertValueCompare >= 0) {
+                // child HV >= target HV
+                if (hilbertValueCompare == 0) {
+                    // child == target
+                    final var tupleCompare = TupleHelpers.compare(slot.getKey(), key);
+                    if (tupleCompare >= 0) {
+                        if (tupleCompare == 0) {
+                            return -1;
+                        }
+                        return slotIndex;
+                    }
+                } else {
+                    return slotIndex;
+                }
+            }
+        }
+
+        return leafNode.size();
+    }
+
+    private static int findDeleteItemSlotIndex(@Nonnull final LeafNode leafNode,
+                                               @Nonnull final BigInteger hilbertValue,
+                                               @Nonnull final Tuple key) {
+        final List<ItemSlot> items = leafNode.getItems();
+        for (int slotIndex = 0; slotIndex < items.size(); slotIndex++) {
+            final ItemSlot slot = items.get(slotIndex);
+            final int hilbertValueCompare = slot.getHilbertValue().compareTo(hilbertValue);
+            if (hilbertValueCompare >= 0) {
+                // child HV >= target HV
+                if (hilbertValueCompare == 0) {
+                    final var tupleCompare = TupleHelpers.compare(slot.getKey(), key);
+                    if (tupleCompare >= 0) {
+                        if (tupleCompare == 0) {
+                            return slotIndex;
+                        }
+                        return -1;
+                    }
+                } else {
+                    return -1;
+                }
+            }
+        }
+
+        return -1;
+    }
+
+    @Nonnull
+    private static Rectangle computeMbr(@Nonnull final List<? extends NodeSlot> slots) {
```

Comment:
I'm not quite sure where to put this, but in out of band discussions, I believe you discussed how you were thinking about using some fuzziness around points to decrease the number of intermediate page updates. (That is, expanding the MBR a bit so that more points wind up in pre-existing intermediate nodes.) Is that still something you want to do?

Reply:
Yes -- will need to do this for temporal data as it is usually monotonically increasing. This is not handled yet.